lib/src/styles/macos_theme_data.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:flutter/foundation.dart';
+
+import '../../macos_ui.dart';
+
+/// Styling specifications for a [MacosTheme].
+///
+/// All constructor parameters can be null, in which case a
+/// [CupertinoColors.activeBlue] based default macOS theme styling is used.
+///
+/// See also:
+///
+///  * [MacosTheme], in which this [MacosThemeData] is inserted.
+class MacosThemeData with Diagnosticable {
+  /// Creates a [MacosTheme] styling specification.
+  ///
+  /// Unspecified parameters default to a reasonable macOS default style.
+  factory MacosThemeData({
+    Brightness? brightness,
+    Color? primaryColor,
+    Color? accentColor,
+    Curve? animationCurve,
+    Duration? mediumAnimationDuration,
+    Typography? typography,
+  }) {
+    final Brightness _brightness = brightness ?? Brightness.light;
+    final bool isDark = _brightness == Brightness.dark;
+    primaryColor ??= isDark
```

Comment:
You are not considering when the device is in high contrast mode. `CupertinoDynamicColor` have all the contrast options

Reply:
This is true. I was intending to add it later.

Reply:
Actually, I think this is not necessary, because the color handling should be up to the widget (`CupertinoDynamicColors.resolve(color, context)`). So, you should not be handling if the brightness is dark or not, because `CupertinoDynamicColor` already have the dark color variation. If you want to add dark mode to a color, make it a CupertinoDynamicColor and set the dark property

Reply:
I'm a little confused, would you mind clarifying what you think should/shouldn't be changed?

Reply:
You're checking if the current theme is dark to set a color, but it shouldn't be done this way. `CupertinoDynamicColor` have all the color variants. If it needs to have a dark theme variation, it should be set in `CupertinoDynamicColor`.

Reply:
Ah, gotcha. I remember seeing that flutter/cupertino has a way to handle that. I'll look into it more. That said, I'm inclined to say that not handling high contrast shouldn't be a merge-blocker at this point.